core/consensus/src/lib.rs

```diff
@@ -100,7 +100,69 @@ pub enum ConsensusError {
 
     /// The synchronous block does not pass the checks.
     #[display(fmt = "Synchronization {} block error", _0)]
-    SyncBlockHashErr(u64),
+    VerifyBlockHeaderPreBlockHash(u64),
+
+    /// The synchronous block does not pass the checks.
+    #[display(fmt = "Synchronization {} block prehash error", _0)]
+    VerifyBlockHeaderPreHash(u64),
+
+    #[display(
+        fmt = "Synchronization {} block error, proposer is not in verify list",
+        _0
+    )]
+    VerifyBlockHeaderProposer(u64),
+
+    /// the validator is not in the verify list
+    #[display(
+        fmt = "Synchronization {} block error, proposer is not in verify list",
+        _0
+    )]
+    VerifyBlockHeaderValidator(u64),
```

Comment:
Refactor these errors to 
```rust
VerifyBlockHeaderErr {
    field: BlockHeaderField,
    height: u64,
}
```
and add another `enum BlockHeaderField`.

Reply:
refactored